scripts/update-generator/main.go

```diff
@@ -129,10 +146,23 @@ func createUpdate(targetPath string, channel, version, platform string, installe
 		return errs.Wrap(err, "Could not create temp binary dir")
 	}
 
-	for _, bf := range binaries {
-		err := copyFileToDir(bf, binTempDir, true)
-		if err != nil {
-			return errs.Wrap(err, "Failed to copy binary file %s", bf)
+	sysTempDir := filepath.Join(tempDir, "system")
+	err = os.MkdirAll(sysTempDir, 0755)
+	if err != nil {
+		return errs.Wrap(err, "Could not create temp system dir")
+	}
+
+	for _, f := range files {
+		if fileutils.IsDir(f) {
+			err := copyDirToDir(f, sysTempDir)
+			if err != nil {
+				return errs.Wrap(err, "Failed to copy directory %s", f)
+			}
+		} else {
+			err := copyFileToDir(f, binTempDir, true)
+			if err != nil {
+				return errs.Wrap(err, "Failed to copy binary file %s", f)
+			}
 		}
```

Comment:
This works, but it is rather implicit logic: files are binaries => `./bin` and directories are system files that go in the `./system` directory.  Can we maybe provide the system files as a separate argument?  Then we would need to introduce a flag for this, something like `-systemApplicationDir`...